spring-cloud-stream-binder-rabbit/src/test/java/org/springframework/cloud/stream/binder/rabbit/RabbitBinderCleanerTests.java

```diff
@@ -136,10 +136,19 @@ private void waitForConsumerStateNot(String queueName, int state) throws Interru
 				URI uri = UriComponentsBuilder.fromUriString("http://localhost:15672/api/queues").pathSegment(
 						"{vhost}", "{queue}")
 						.buildAndExpand("/", queueName).encode().toUri();
-				while (n++ < 100) {
-					@SuppressWarnings("unchecked")
+
+				Object consumers = null;
+				while (n++ < 100 && consumers == null) {
 					Map<String, Object> queueInfo = template.getForObject(uri, Map.class);
-					if (!queueInfo.get("consumers").equals(Integer.valueOf(state))) {
+					consumers = queueInfo.get("consumers");
+					if (consumers == null){
+						Thread.sleep(100);
+					}
+				}
+				assertThat(consumers != null);
+
+				while (n++ < 100) {
+					if (!consumers.equals(Integer.valueOf(state))) {
```

Comment:
We don't need this second loop - nothing changes inside of it.

This needs to go to 144 

    if (consumers == null || consumers.equals(Integer.valueOf(state))) {

I can fix during merge if you agree.